src/Microsoft.AspNet.Mvc.Core/ActionContext.cs

```diff
@@ -13,6 +14,21 @@ namespace Microsoft.AspNet.Mvc
     public class ActionContext
     {
         /// <summary>
+        /// Creates a empty <see cref="ActionContext"/>.
+        /// </summary>
+        /// <remarks>
+        /// The default constructor should be used only when <see cref="ActionContext"/> needs to be directly
+        /// instantiated in user codes, such as for unit tests.
+        /// </remarks>
+        public ActionContext()
```

Comment:
This comment isn't clear enough. Just say `This constructor is provided for unit testing purposes`.